components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAOTest.java

```diff
@@ -0,0 +1,142 @@
+    private static final String appState = "ACTIVE";
+    private static final String callback = "http://localhost:8080/redirect";
+    private static final String DB_Name = "testDB";
+
+    @Mock
+    private OAuthServerConfiguration mockedServerConfig;
+
+    @BeforeClass
+    public void setUp() throws Exception {
+
+        initiateH2Base(DB_Name, getFilePath("h2.sql"));
+        createBase(clientId, secret, username, appName, callback, appState);
+    }
+
+    @Test
+    public void testRemoveConsumerApplication() throws Exception {
+
+        String GET_SECRET = "SELECT CONSUMER_SECRET FROM IDN_OAUTH_CONSUMER_APPS WHERE CONSUMER_KEY=?";
+        String secret = null;
+
+        PreparedStatement statement = null;
+        ResultSet resultSet = null;
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mockedServerConfig);
+        PlainTextPersistenceProcessor processor = new PlainTextPersistenceProcessor();
+        when(mockedServerConfig.getPersistenceProcessor()).thenReturn(processor);
+
+        try(Connection connection = getConnection(DB_Name)) {
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+
+            OAuthAppDAO AppDAO = new OAuthAppDAO();
+            AppDAO.removeConsumerApplication(clientId);
+
+            try {
+                statement = connection.prepareStatement(GET_SECRET);
+                statement.setString(1, clientId);
+                resultSet = statement.executeQuery();
+                if(resultSet.next()) {
+                    secret = resultSet.getString(1);
+                }
+            } finally {
+                IdentityDatabaseUtil.closeAllConnections(connection, resultSet, statement);
+            }
+            assertNull(secret, "Checking whether the CONSUMER_SECRET is successfully deleted.");
+        }
+    }
+
+    @Test
+    public void testUpdateOAuthConsumerApp() throws Exception {
+
+        String GET_APP = "SELECT APP_NAME FROM IDN_OAUTH_CONSUMER_APPS WHERE CONSUMER_KEY=?";
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(mockedServerConfig);
+        PlainTextPersistenceProcessor processor = new PlainTextPersistenceProcessor();
+        when(mockedServerConfig.getPersistenceProcessor()).thenReturn(processor);
+
+        try(Connection connection1 = getConnection(DB_Name);
+            PreparedStatement statement = connection1.prepareStatement(GET_APP)) {
+
+            mockStatic(IdentityDatabaseUtil.class);
+            when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection1);
+
+            OAuthAppDAO AppDAO = new OAuthAppDAO();
+            AppDAO.updateOAuthConsumerApp(appName, clientId);
+            statement.setString(1, clientId);
+            try(ResultSet resultSet = statement.executeQuery()) {
+                if(resultSet.next()) {
```

Comment:
what if resultSet.next is false ? Will this test pass ? Is that expected ?

Reply:
Hopefully, it won't be false. Since I am populating the table in the setup method the statement will always return required values. Please correct me if I'm wrong.